components/core/src/utils/make_dictionaries_readable/make-dictionaries-readable.cpp

```diff
@@ -95,15 +107,25 @@ int main (int argc, const char* argv[]) {
 
     // Write readable dictionary
     auto readable_var_dict_path = boost::filesystem::path(command_line_args.get_output_dir()) / streaming_archive::cVarDictFilename;
+    auto readable_var_dict_index_path = boost::filesystem::path(command_line_args.get_output_dir()) / streaming_archive::cVarSegmentIndexFilename;
     readable_var_dict_path += ".hr";
+    readable_var_dict_index_path += ".hr";
     file_writer.open(readable_var_dict_path.string(), FileWriter::OpenMode::CREATE_FOR_WRITING);
+    index_writer.open(readable_var_dict_index_path.string(), FileWriter::OpenMode::CREATE_FOR_WRITING);
     for (const auto& entry : var_dict.get_entries()) {
         file_writer.write_string(entry.get_value());
         file_writer.write_char('\n');
+        std::set<segment_id_t> segment_ids_set = entry.get_ids_of_segments_containing_entry();
+        // the ids_set is std::set which iterates id in ascending order
+        for (const auto segment_id : segment_ids_set) {
+            index_writer.write_string(std::to_string(segment_id) + " ");
+        }
+        index_writer.write_char('\n');
     }
     file_writer.close();
+    index_writer.close();
 
     var_dict.close();
 
     return 0;
-}
+}
```

Comment:
It's a nicety to leave a newline at the end of the file, especially when working with command-line tools, so I would leave it in.